packages/devtools_app/integration_test/test_infra/run/run_test.dart

```diff
@@ -315,3 +316,45 @@ class TestRunnerArgs {
   /// instead of 'chrome'.
   late final bool headless;
 }
+
+enum TestAppDevice {
+  flutterTester,
+  chrome;
+
+  /// A mapping of test app device to the unsupported tests for that device.
+  static final _unsupportedTestsForDevice = <TestAppDevice, List<String>>{
+    TestAppDevice.flutterTester: [],
+    TestAppDevice.chrome: [
+      // TODO(https://github.com/flutter/devtools/issues/5874): Remove once supported on web.
+      'eval_and_browse_test.dart',
+      'perfetto_test.dart',
+      'performance_screen_event_recording_test.dart',
+      'service_connection_test.dart',
+    ],
+  };
+
+  static final _argNameToDeviceMap =
+      TestAppDevice.values.fold(<String, TestAppDevice>{}, (map, device) {
+    map[device.argName] = device;
+    return map;
+  });
+
+  static TestAppDevice? fromArgName(String argName) {
+    return _argNameToDeviceMap[argName];
+  }
+
+  String get argName {
+    switch (this) {
+      case flutterTester:
+        return 'flutter-tester';
+      case chrome:
+        return 'chrome';
+    }
+  }
```

Comment:
instead of making this a getter, we can have a field `argName` that is set from a constructor 

```dart
enum TestAppDevice {
  flutterTester('flutter-tester'),
  chrome('chrome');

  const TestAppDevice(this.argName);

  final String argName;
```

Reply:
Oooh I didn't know that was possible with enums! Switched to setting it from a constructor